packages/x-data-grid/src/hooks/features/editing/useGridRowEditing.ts

```diff
@@ -433,7 +433,28 @@ export const useGridRowEditing = (
 
         let newValue = apiRef.current.getCellValue(id, field);
         if (fieldToFocus === field && (deleteValue || initialValue)) {
-          newValue = deleteValue ? '' : initialValue;
+          if (deleteValue) {
```

Comment:
should this be extracted in a small util (since it is identical to the code in the cell editing hook)

Reply:
Might be a good idea

Reply:
I did now create a new API method `getDefaultCellValue`, since I think that this might be used elsewhere or in userland as well. WDYT @arminmeh ?

Reply:
What I saw so far is that we keep those private until they are actually needed.

Reply:
we can do that as well ... but I don't see any harm in making it public rn ... thoughts @mui/xgrid ?

Reply:
I'd keep it (`getDefaultCellValue`) inline if it does not need to be reused.
Or an unexported internal function of the `useGridRowEditing` hook, if there's a need to reuse it internally. (Or possibly a generic utility function outside the component scope that receives `colDef` as a parameter)
Later, if a need arises in the userland for some use case that is not possible otherwise, would consider exporting it.

The motive is to keep the API as simple as possible since it would make life easier for the team (fewer breaking changes) and also for the users (simpler product).